### PR TITLE
made changes to the documentaiton of developer in python3 print

### LINKED ISSUE #36807: developer's guide still mentions python2/3
### Steps To Reproduce

https://doc.sagemath.org/html/en/developer/coding_basics.html#writing-testable-examples mentions
```
Python3 print: Python3 syntax for print must be used in Sage code and doctests. If you use an old-style print in doctests, it will raise a SyntaxError:
```
I think this can be removed now.

### Expected Behavior

python 2 support has been removed long ago

### Actual Behavior

python 2 print is mentioned

### Additional Information

_No response_

### Environment

```markdown
irrelevant
```

### Checklist

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.
- [X] I have read the documentation and troubleshoot guide

### PR DESCRIPTION
**Pull request by amanmoon**
removed the old python2 print statment information from the developer documentation

Fixes https://github.com/sagemath/sage/issues/36807

removed the "Python3 print: Python3 syntax for print must be used in Sage code and doctests. If you use an old-style print in doctests, it will raise a SyntaxError:"
and changed it with
"In SageMath code and doctests, adhere to Python 3
syntax for the print function. Use parentheses as follows:"

### CODE REVIEW

File: src/doc/en/developer/coding_basics.rst [hunk 1]
```diff
@@ -1039,14 +1039,9 @@ written.
...
-- **Python3 print:** Python3 syntax for print must be used in Sage
-  code and doctests. If you use an old-style print in doctests, it
-  will raise a SyntaxError::
-
-      sage: print "not like that"
-      Traceback (most recent call last):
-      ...
-      SyntaxError: ...
+- **Python3 print:** In SageMath code and doctests, adhere to Python 3
+  syntax for the print function. Use parentheses as follows:::
```
Comment (dimpase) [inline src/doc/en/developer/coding_basics.rst]: why do you need `:::` there? Maybe `::` should do?

Comment (amanmoon) [reply src/doc/en/developer/coding_basics.rst]: I removed the `:` from end of the line.

Comment (mantepse) [reply src/doc/en/developer/coding_basics.rst]: I would be more in favour of removing the sentence and the doctest completely, don't you agree @dimpase?  There is already so much to read and I don't think that anybody will be assuming python 2 syntax nowadays.

Comment (dimpase) [reply src/doc/en/developer/coding_basics.rst]: good idea. It's just very basic python.

Comment (amanmoon) [reply src/doc/en/developer/coding_basics.rst]: @mantepse & @dimpase so should i move forward with removing the whole print sentence and doctest?

Comment (dimpase) [reply src/doc/en/developer/coding_basics.rst]: yes please

**Comment by mantepse** [comment]
@adrinospy, can you add the label "s:needs review"?  (you may not have the appropriate rights, I don't know)

**Comment by adrinospy** [comment]
> @adrinospy, can you add the label "s:needs review"? (you may not have the appropriate rights, I don't know)

No, no option is there to add label.

**Review by mantepse — APPROVED** [review]
LGTM

**Pull request merged**